spikeinterface/extractors/neoextractors/spikeglx.py

```diff
@@ -37,9 +37,14 @@ def __init__(self, folder_path, stream_id=None):
         NeoBaseRecordingExtractor.__init__(self, stream_id=stream_id, **neo_kwargs)
 
         #~ # open the corresponding stream probe
-        if HAS_NEO_10_2 and '.ap' in self.stream_id:
+        if HAS_NEO_10_2 and '.nidq' not in self.stream_id:
             signals_info_dict = {e['stream_name'] : e for e  in self.neo_reader.signals_info_list}
             meta_filename = signals_info_dict[self.stream_id]['meta_file']
+            # Load probe geometry if available
+            try:
```

Comment:
I would prefer to avoid abusive try except here.
Maybe we could have an option for grouping in the extractor.
Like `group_by_shank=True/False` and if it is True AND the probe have sevral shank we use the `group_mode='by_shank'`

Reply:
or not throw an error is `by_shank` but there are no shank ids...